provider/vsphere/provider_test.go

```diff
@@ -107,3 +112,83 @@ regions:
 	err = p.CloudSchema().Validate(v)
 	c.Assert(err, jc.ErrorIsNil)
 }
+
+type pingSuite struct {
+	testing.IsolationSuite
+}
+
+var _ = gc.Suite(pingSuite{})
+
+func (pingSuite) TestPingInvalidHost(c *gc.C) {
+	tests := []string{
+		"foo.com",
+		"http://IHopeNoOneEverBuysThisVerySpecificJujuDomainName.com",
+		"http://IHopeNoOneEverBuysThisVerySpecificJujuDomainName:77",
+	}
+
+	provider, err := environs.Provider("vsphere")
+	c.Assert(err, jc.ErrorIsNil)
+
+	for _, t := range tests {
+		err := provider.Ping(t)
+		if err == nil {
+			c.Errorf("ping %q: expected error, but got nil.", t)
+			continue
+		}
+		expected := "No VSphere server running at " + t
+		if err.Error() != expected {
+			c.Errorf("ping %q: expected %q got %v", t, expected, err)
+		}
+	}
+}
+
+func (pingSuite) TestPingInvalidURL(c *gc.C) {
+	provider, err := environs.Provider("vsphere")
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = provider.Ping("abc%sdef")
+	if err == nil {
```

Comment:
Yeah, I think I used to have more logic inside the if statement.  But you're right, it's extraneous now.